### PR TITLE
nf-core modules update --all

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/prepare_genome.nf [hunk 1]
```diff
@@ -137,19 +154,15 @@ workflow BUILD_INDICES {
...
+        dbsnp_tbi             = result_dbsnp_tbi
+        dict                  = result_dict
+        fai                   = result_fai
+        germline_resource_tbi = result_germline_resource_tbi
+        intervals             = result_intervals
+        known_indels_tbi      = result_known_indels_tbi.collect()
+        msisensorpro_scan     = result_msisensorpro_scan
+        pon_tbi               = result_pon_tbi
+        target_bed_gz_tbi     = result_target_bed
+        versions              = ch_versions.ifEmpty(null) // channel: [ versions.yml ]
```
Comment (mahesh-panchal) [inline subworkflows/local/prepare_genome.nf:167]: why send `null` into the channel here?

File: subworkflows/local/prepare_genome.nf [hunk 2]
```diff
@@ -44,78 +44,95 @@ workflow BUILD_INDICES {
...
 
-    result_msisensorpro_scan  = Channel.empty()
-    version_msisensorpro_scan = Channel.empty()
-    if ('msisensorpro' in tools)
-        (result_msisensorpro_scan, version_msisensorpro_scan) = MSISENSORPRO_SCAN(fasta)
+    result_msisensorpro_scan = Channel.empty()
+    if ('msisensorpro' in tools) {
+        MSISENSORPRO_SCAN(fasta)
+        result_msisensorpro_scan = MSISENSORPRO_SCAN.out.list
+        ch_versions = ch_versions.mix(MSISENSORPRO_SCAN.out.versions)
```
Comment (mahesh-panchal) [inline subworkflows/local/prepare_genome.nf:125]:
```suggestion
        ch_versions.mix(MSISENSORPRO_SCAN.out.versions)
```

File: subworkflows/local/prepare_genome.nf [hunk 3]
```diff
@@ -44,78 +44,95 @@ workflow BUILD_INDICES {
...
 
-    result_pon_tbi  = Channel.empty()
-    version_pon_tbi = Channel.empty()
+    result_pon_tbi = Channel.empty()
     if (!(params.pon_tbi) && params.pon && ('tnscope' in tools || 'mutect2' in tools)) {
         pon_id = pon.map{ it -> [[id:it[0].baseName], it] }
-        (result_pon_tbi, version_pon_tbi) = TABIX_PON(pon_id)
+        TABIX_PON(pon_id)
+        result_pon_tbi = TABIX_PON.out.tbi.map{ meta, tbi -> [pon, tbi] }
+        ch_tabix_versions = ch_tabix_versions.mix(TABIX_PON.out.versions).first()
```
Comment (mahesh-panchal) [inline subworkflows/local/prepare_genome.nf:116]:
```suggestion
        ch_versions.mix(TABIX_PON.out.versions.first())
```
Same with all the others

File: subworkflows/local/prepare_genome.nf [hunk 4]
```diff
@@ -44,78 +44,95 @@ workflow BUILD_INDICES {
...
 
-    result_known_indels_tbi  = Channel.empty()
-    version_known_indels_tbi = Channel.empty()
+    result_known_indels_tbi = Channel.empty()
     if (!(params.known_indels_tbi) && params.known_indels && ('mapping' in step || 'preparerecalibration' in step)) {
         known_indels_id = known_indels.map{ it -> [[id:it[0].baseName], it] }
-        (result_known_indels_tbi, version_known_indels_tbi) = TABIX_KNOWN_INDELS(known_indels_id)
-        result_known_indels_tbi = result_known_indels_tbi.map{ meta, tbi -> [tbi] }
+        TABIX_KNOWN_INDELS(known_indels_id)
+        result_known_indels_tbi = TABIX_KNOWN_INDELS.out.tbi.map{ meta, tbi -> [known_indels, tbi] }
```
Comment (mahesh-panchal) [inline subworkflows/local/prepare_genome.nf:107]:
```suggestion
        result_known_indels_tbi.mix(TABIX_KNOWN_INDELS.out.tbi.map{ meta, tbi -> [known_indels, tbi] })
```
Does this work? `known_indels` should be a channel, not a data element.

Comment (mahesh-panchal) [reply subworkflows/local/prepare_genome.nf:107]: It seems it's implicitly turned into a channel only in the process call, not in `take:`.

File: subworkflows/local/prepare_genome.nf (hunk 1 above)
Comment (maxulysse) [reply subworkflows/local/prepare_genome.nf]: Copied that over from the rnaseq pipeline

File: subworkflows/local/prepare_genome.nf (hunk 4 above)
Comment (maxulysse) [reply subworkflows/local/prepare_genome.nf]: It is working, we tried that with @FriederikeHanssen
But we might have to do some adjustments later

**Comment by maxulysse** [comment]
Finally manage to figure out the issue...
Long story short, be careful when updating modules, especially when you change a local for a nf-core one.

**Pull request closed without merging**